### ISSUE TITLE
svg support branch

### ISSUE THREAD

**Issue by frostidaho** [qtile/qtile#1014]
## SVG support branch

I've created a svg support for some widgets and 
refactored how images are loaded for widgets in general.

Qtile's widgets deal with images by creating cairo `cairocffi.ImageSurface` and `cairocffi.SurfacePattern` objects which correspond to some image.

This branch refactors loading and resizing images into [libqtile/images.py](https://github.com/frostidaho/qtile/blob/b0f929f76d43d6e23ee5ca893d695165677da53e/libqtile/images.py). The `libqtile.images.Loader` class can load png and optionally svg files.
For svg support you need to have [cairosvg](https://pypi.python.org/pypi/CairoSVG) installed.

The [develop_svg_support](https://github.com/frostidaho/qtile/tree/develop_svg_support) branch also has some new / slightly modified widgets.

[VolumeImg](https://github.com/frostidaho/qtile/blob/develop_svg_support/libqtile/widget/volumeimg.py) is a
widget in this branch which shows how `libqtile.images.Loader` is used. There are also a number of tests
which show how `Loader` works.

* [example bar with png & svg icons](http://i.imgur.com/fkSWRQM.png)
* [example bar code](https://gist.github.com/frostidaho/88778b1926dc44f14c309a01cb849777)

If there is interest in it, I'd be happy to clean it up.

**Comment by flacjacket** [comment]
I think adding support for svgs in widgets would be really great. I have hacked in svg support into some things for myself, but it was really rough, lots of manual svg drawing, and I eventually gave up and found fontawesome icons that did what I needed. This would definitely open up new avenues for building widgets. There are even things that we already have like the layout icons that are png and would be great in svg. I'd definitely be +1
